src/context.rs

```diff
@@ -206,71 +110,23 @@ impl<A: Actor> Context<A> {
 
     /// Handle a message and immediate notifications, returning whether to exit from the manage loop
     /// or not.
-    async fn tick(
-        &mut self,
-        msg: Either<BroadcastMessage<A>, AddressMessage<A>>,
-        actor: &mut A,
-    ) -> ControlFlow<()> {
+    async fn tick(&mut self, msg: ActorMessage<A>, actor: &mut A) -> ControlFlow<()> {
         match msg {
-            Either::Left(BroadcastMessage::Message(msg)) => msg.handle(actor, self).await,
-            Either::Left(BroadcastMessage::Shutdown) => {
-                self.running = false;
-                return ControlFlow::Break(());
-            }
-            Either::Right(AddressMessage::Message(msg)) => {
-                msg.handle(actor, self).await;
-            }
-            Either::Right(AddressMessage::LastAddress) => {
-                if self.ref_counter.strong_count() == 0 {
-                    self.stop_all();
-                    self.running = false;
-                    return ControlFlow::Break(());
-                }
-            }
+            ActorMessage::ToAllActors(msg) => msg.handle(actor, self).await,
+            ActorMessage::Shutdown => self.running = false,
+            ActorMessage::ToOneActor(msg) => msg.handle(actor, self).await,
```

Comment:
```suggestion
            ActorMessage::ToOneActor(msg) => msg.handle(actor, self).await,
            ActorMessage::ToAllActors(msg) => msg.handle(actor, self).await,
            ActorMessage::Shutdown => self.running = false,
```

I'd mildly prefer this order.